src/SignalRServiceExtension/Config/DependencyInjectionExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal static class DependencyInjectionExtensions
+    {
+        private const string HubProtocolError = "It's invalid to configure hub protocol on Azure Functions runtime V2. Newtonsoft.Json protocol will be used.";
+
+        public static IServiceCollection SetHubProtocol(this IServiceCollection services, IConfiguration configuration)
+        {
+            if (Environment.Version.Major == 4)
+            {
+                // Actually is .Net Core 2.x
+                if (configuration[Constants.AzureSignalRHubProtocol] != null)
+                {
+                    throw new InvalidOperationException(HubProtocolError);
+                }
+            }
+            else
+            {
+#if NETCOREAPP3_1
+                if (Not_DotnetRuntime(configuration) || UserSpecifyNewtonsoft(configuration))
+                {
+                    //this would overwrite the System.Text.Json Protocol.
+                    services.TryAddEnumerable(ServiceDescriptor.Singleton<IHubProtocol, NewtonsoftJsonHubProtocol>());
+                }
+#endif
+            }
+            return services;
+        }
+
+#if NETCOREAPP3_1
```

Comment:
remove the framework check? too much if.

Reply:
`NewtonsoftJsonHubProtocol` only exists in .Net 3.0 +

Reply:
how about move this simple method into above function, to reduce `#if`?

Reply:
Then it would become:
```cs
var userSpecifyNewtonsoft = configuration.GetValue(Constants.AzureSignalRHubProtocol, HubProtocol.SystemTextJson) == HubProtocol.NewtonsoftJson;

if (Not_DotnetRuntime(configuration) || userSpecifyNewtonsoft )
                {
                    //this would overwrite the System.Text.Json Protocol.
                    services.TryAddEnumerable(ServiceDescriptor.Singleton<IHubProtocol, NewtonsoftJsonHubProtocol>());
                }
```
Seems not symmetrical and less readability? Also no need to compile the code in the second `#if` .